homeassistant/components/instructure/calendar.py

```diff
@@ -33,35 +33,27 @@ def __init__(self, hass, coordinator, entity_id):
     @property
     def event(self) -> CalendarEvent | None:
         """Return the next upcoming event."""
-        assignments = self.get_assignments_with_due_date()
+        assignments = self.coordinator.data[ASSIGNMENTS_KEY].values()
 
         if not assignments:
             return None
 
-        current_time = datetime.datetime.now(timezone.utc)
-
         next_assignment = min(
             (
                 item
                 for item in assignments
-                if self.parse_date(item["due_at"]) > current_time
             ),
             key=lambda item: self.parse_date(item["due_at"]),
             default=None,
         )
```

Comment:
Since you fixed this one, cant you write it like this?

next_assignment = min(assignments, key=lambda item: self.parse_date(item["due_at"]), default=None)

Reply:
fixed